pkg/kubelet/metrics/metrics.go

```diff
@@ -211,6 +213,30 @@ var (
 		[]string{},
 	)
 
+	// FirstNetworkPodStartSLIDuration is a gauge that tracks the duration (in seconds) it takes for the first network pod to run,
+	// excluding the time for image pulling. This metric should reflect the "Pod startup latency SLI" definition and measure the time
+	// for the CNI plugin to be able to provide the network capabilities to the Pod.
+	// ref: https://github.com/kubernetes/community/blob/master/sig-scalability/slos/pod_startup_latency.md
+	FirstNetworkPodStartSLIDuration = metrics.NewGauge(
+		&metrics.GaugeOpts{
+			Subsystem:      KubeletSubsystem,
+			Name:           FirstNetworkPodStartSLIDurationKey,
+			Help:           "Duration in seconds to start the first network pod, excluding time to pull images and run init containers, measured from pod creation timestamp to when all its containers are reported as started and observed via watch",
+			StabilityLevel: metrics.ALPHA,
+		},
+	)
+
+	// FirstNetworkPodStartTotalDuration is a gauge that tracks the duration (in seconds) it takes for the first network pod to run
+	// since creation, including the time for image pulling.
+	FirstNetworkPodStartTotalDuration = metrics.NewGauge(
+		&metrics.GaugeOpts{
+			Subsystem:      KubeletSubsystem,
+			Name:           FirstNetworkPodStartTotalDurationKey,
+			Help:           "Duration in seconds to start the first network pod since creation, including time to pull images and run init containers, measured from pod creation timestamp to when all its containers are reported as started and observed via watch",
+			StabilityLevel: metrics.ALPHA,
+		},
+	)
```

Comment:
These are very strange metrics.. are you basically just recording this a single time?

Reply:
yes, the goal is to measure the user perceived latency when running workloads, we already have node startup latency https://github.com/kubernetes/kubernetes/pull/118568 and pod startup latency, but the latency perceived by an user will  be the "node startup + first pod with network latency"

Reply:
@aojea is this the only place at which we can measure CNI plugin initialization? Because measuring it at the pod startup level kind of pollutes the data with other variables such as the image pulling latency.